frontends/main/src/app-pages/DashboardPage/CoursewareDisplay/types.ts

```diff
@@ -0,0 +1,52 @@
+const DashboardResourceType = {
+  Course: "course",
+  Program: "program",
+} as const
+type DashboardResourceType =
+  (typeof DashboardResourceType)[keyof typeof DashboardResourceType]
+
+const EnrollmentStatus = {
+  NotEnrolled: "not_enrolled",
+  Enrolled: "enrolled",
+  Completed: "completed",
+} as const
+type EnrollmentStatus = (typeof EnrollmentStatus)[keyof typeof EnrollmentStatus]
+
+const EnrollmentMode = {
+  Audit: "audit",
+  Verified: "verified",
+} as const
+type EnrollmentMode = (typeof EnrollmentMode)[keyof typeof EnrollmentMode]
+
+type DashboardCourse = {
```

Comment:
Besides the renaming  this (Previously it was `EnrollmentData`):
- run-derived properties are now nested under run
- enrollment-derived properties are now nested under enrollment
- `hasCompleted` boolean replaced by `EnrollmentStatus = "not_enrolled" | "enrolled" | "completed".